### PR TITLE
Fix Tensor.uniform_ documentation to mention generator argument

### LINKED ISSUE #98202: `Tensor.uniform_` misses `generator` arg
### 📚 The doc issue

https://pytorch.org/docs/stable/generated/torch.Tensor.uniform_.html

It just specifies the signature `Tensor.uniform_(from=0, to=1) → Tensor` but actually it also supports the `generator` argument.

### Suggest a potential alternative/fix

Add `generator` to the arguments.

cc @svekars @carljparker @pbelevich

### PR DESCRIPTION
**Pull request by tfsingh**
Fixes #98202 , mentioning support for generator arguments while providing an example of the function's use.

Would be happy to iterate on this if there are any issues.

cc @soulitzer

### CODE REVIEW

File: torch/_tensor_docs.py [hunk 1]
```diff
@@ -5937,13 +5937,29 @@ def callable(a, b) -> number
 add_docstr_all(
     "uniform_",
     r"""
-uniform_(from=0, to=1) -> Tensor
+uniform_(from, to, generator) -> Tensor
```
Comment (soulitzer) [inline torch/_tensor_docs.py:5940]: I would change to: `uniform_(from=0, to=1, *, generator=None) -> Tensor`
- Keep the default values?
- We should indicate generator is a kwarg-only argument

File: torch/_tensor_docs.py [hunk 2]
```diff
@@ -5937,13 +5937,29 @@ def callable(a, b) -> number
...
 
 Fills :attr:`self` tensor with numbers sampled from the continuous uniform
 distribution:
 
 .. math::
     P(x) = \dfrac{1}{\text{to} - \text{from}}
+
+Args:
+    from (float): lower-bound of the range to sample from
```
Comment (soulitzer) [inline torch/_tensor_docs.py:5949]: To keep this PR simple, I'd suggest just removing this section for now as it doesn't seem to add new information like whether the bounds are inclusive are not. https://github.com/pytorch/pytorch/issues/16706

**Comment by tfsingh** [comment]
Of course, these edits should address your comments.

**Review by soulitzer — APPROVED** [review]
Thanks!

**Pull request closed without merging**